frontend/catalyst/jax_primitives.py

```diff
@@ -408,10 +409,15 @@ def _grad_lowering(ctx, *args, jaxpr, fn, grad_params):
     # ``ir.DenseElementsAttr.get()`` constructs a dense elements attribute from an array of
     # element values. This doesn't support ``jaxlib.xla_extension.Array``, so we have to cast
     # such constants to numpy array types.
-    constants = [
-        StableHLOConstantOp(ir.DenseElementsAttr.get(np.asarray(const))).results
-        for const in jaxpr.consts
-    ]
+
+    constants = []
+    for const in jaxpr.consts:
+        const_type = shape_dtype_to_ir_type(const.shape, const.dtype)
+        constants.append(
+            StableHLOConstantOp(
+                ir.DenseElementsAttr.get(np.asarray(const), type=const_type)
+            ).results
+        )
```

Comment:
I prefer statements in a single line if possible

```suggestion
        const_type = shape_dtype_to_ir_type(const.shape, const.dtype)
        nparray = np.asarray(const)
        attr = ir.DenseElementsAttr.get(nparray, type=const_type)
        constantVals = StableHLOConstantOp(attr).results
        constants.append(constantVals)
```

Reply:
done!